### DISCUSSION TITLE
Permission denied when running ./pocketbase serve on ubuntu server 22.04

### DISCUSSION THREAD

**Discussion by altacountbabi** [pocketbase/pocketbase#2115] [Q&A] [answered]
I tried running it with and without root and I have no idea what could be causing this, I transferred it from my laptop with scp

**Comment by ganigeorgiev** [comment, marked as answer]
Could you share the exact error message and command that you've tried running?

Make sure that your os user have read/write access of the directory where the PocketBase executable is located. You can also specify an explicit `pb_data` dir location with the `--dir='/path/to/my/pb_data'` flag (_for list of all supported flags you can use `--help`_).

**Reply by altacountbabi** [reply]
![image](https://user-images.githubusercontent.com/82091823/226734096-34ff9a2a-b5cc-483b-b03f-baa846a4bf52.png)
I actually can't run it at all, its inside /home/username/test and I have read/write access

**Reply by ganigeorgiev** [reply]
Hm, maybe the `pocketbase` file is missing "execution" permission. You can try running `sudo chmod +x  pocketbase` to give "execution" permission to every user (_all `chmod u+x pocketbase` for only the current user_).

You can verify the executable permissions with `ls -la` (if there is `x` then you should be able to run it).

**Reply by altacountbabi** [reply]
`sudo chmod +x  pocketbase` worked, thank you!

**Reply by HMEsau** [reply]
Hi there, I also have the same problem. Mine say Exec format error. SOS

**Reply by altacountbabi** [reply]
Maybe the binary got corrupted? Try redownloading it

**Comment by Khaled-BH** [comment]
I've done the preiv steps and still has same issue 

Mar 05 23:02:42 localhost.localdomain systemd[1438]: pocketbase.service: Failed to set up standard output: Permission denied
Mar 05 23:02:42 localhost.localdomain systemd[1438]: pocketbase.service: Failed at step STDOUT spawning /usr/local/bin/pb/pocketbase: Permission denied

**Reply by ganigeorgiev** [reply]
@Khaled-BH Without details about your systemd service file and your environment I'm not sure how to help you.

Since the errors are related to the std output set up, if you have `StandardError` and `StandardOutput` systemd service instructions make sure that the specified log files directory exists and is writable.

**Reply by aleBranching** [reply]
I ran into the same issue. Most likely you are running a distribution with SELinux.
you can check if it's the issue by disabling it `sudo setenforce 0` and then `sudo systemctl restart pocketbase`

to make it run with SELinux you have to:

put the pocketbase in root directory. I'm not a 100% sure you do need to but SELinux seems to be funny about you running systemd services execstart from home directory. 
change the output logging. It seems to be funny about appending to files
```
StandardOutput = journal
StandardError  = journal
```

then change the execution context of pocketbase
 sudo chcon -t bin_t /pb/pocketbase
 
 For clarity here's what the unit file should look like 
 
 ```
[Unit]
Description = pocketbase

[Service]
Type           = simple
User           = root
Group          = root
LimitNOFILE    = 4096
Restart        = always
RestartSec     = 5s
StandardOutput = journal
StandardError  = journal
ExecStart      = /pb/pocketbase serve --http="172.236.21.185:80"

[Install]
WantedBy = multi-user.target
```
